operator/helper/operatortest/operatortest.go

```diff
@@ -0,0 +1,87 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operatortest
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+	"testing"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+
+	"github.com/open-telemetry/opentelemetry-log-collection/operator/helper"
+)
+
+// ConfigTestCase is used for testing golden configs
+type ConfigTestCase struct {
+	Name      string
+	Expect    interface{}
+	ExpectErr bool
+}
+
+func configFromFileViaYaml(file string, config interface{}) (interface{}, error) {
+	bytes, err := ioutil.ReadFile(file)
+	if err != nil {
+		return nil, fmt.Errorf("could not find config file: %s", err)
+	}
+	if err := yaml.Unmarshal(bytes, config); err != nil {
+		return nil, fmt.Errorf("failed to read config file as yaml: %s", err)
+	}
+
+	return config, nil
+}
+
+func configFromFileViaMapstructure(file string, config interface{}) (interface{}, error) {
+	bytes, err := ioutil.ReadFile(file)
+	if err != nil {
+		return nil, fmt.Errorf("could not find config file: %s", err)
+	}
+
+	raw := map[string]interface{}{}
+
+	if err := yaml.Unmarshal(bytes, raw); err != nil {
+		return nil, fmt.Errorf("failed to read data from yaml: %s", err)
+	}
+
+	dc := &mapstructure.DecoderConfig{Result: config, DecodeHook: helper.JSONUnmarshalerHook()}
+	ms, err := mapstructure.NewDecoder(dc)
+	if err != nil {
+		return nil, err
+	}
+	err = ms.Decode(raw)
+	if err != nil {
+		return nil, err
+	}
+	return config, nil
+}
+
+// RunGoldenConfigTest Unmarshalls yaml files and compares them against the expected.
+func RunGoldenConfigTest(t *testing.T, config interface{}, tc ConfigTestCase) {
```

Comment:
Let's change this to `func (c ConfigUnmarshalTest) Run(t *testing.T, config interface{}) {`